File: dom-server/src/main/java/com/dorm/server/entity/Dormitory.java
package com.dorm.server.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 宿舍实体类
 * 映射 dormitories 表
 *
 * @author dorm-server
 */
@Data
public class Dormitory {

    /** 主键ID */
    private Integer id;

    /** 所属地域ID */
    private Integer regionId;

    /** 宿舍名称 */
    private String name;

    /** 宿舍类型（male/female/mixed） */
    private String dormitoryType;

    /** 地址 */
    private String address;

    /** 日额费用 */
    private BigDecimal dailyRate;

    /** 排序号 */
    private Integer sortOrder;

    /** 乐观锁版本号 */
    private Integer version;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;

    /** 软删除时间（null表示未删除） */
    private LocalDateTime deletedAt;
}

File: dom-dev/src/types/dormitory.ts
export interface Dormitory {
  id: number
  regionId: number
  regionName?: string
  name: string
  address?: string
  dailyRate: number
  sortOrder: number
  dormitoryType: 'male' | 'female' | 'mixed'
  version: number
  createdAt: string
  updatedAt: string
  totalRooms: number
  occupiedRooms: number
}

export interface CreateDormitoryRequest {
  regionId: number
  name: string
  address?: string
  dailyRate: number
  sortOrder: number
}

export interface UpdateDormitoryRequest {
  regionId: number
  name: string
  address?: string
  dailyRate: number
  sortOrder: number
  version: number
}

export interface DormitoryListParams {
  page?: number
  pageSize?: number
  keyword?: string
  regionId?: number
}
